block-producer/src/server/api.rs

```diff
@@ -63,23 +167,36 @@ where
 {
     async fn submit_proven_transaction(
         &self,
-        _request: tonic::Request<SubmitProvenTransactionRequest>,
+        request: tonic::Request<SubmitProvenTransactionRequest>,
     ) -> Result<tonic::Response<SubmitProvenTransactionResponse>, Status> {
-        todo!()
+        let request = request.into_inner();
+
+        let tx = ProvenTransaction::read_from_bytes(&request.transaction)
+            .map_err(|_| Status::invalid_argument("Invalid transaction"))?;
+
+        self.queue
+            .add_transaction(Arc::new(tx))
+            .await
+            .map_err(|err| Status::invalid_argument(format!("{:?}", err)))?;
+
+        Ok(tonic::Response::new(SubmitProvenTransactionResponse {}))
     }
 }
 
 pub async fn serve(config: BlockProducerConfig) -> Result<()> {
     let host_port = (config.endpoint.host.as_ref(), config.endpoint.port);
     let addrs: Vec<_> = host_port.to_socket_addrs()?.collect();
 
-    let store = Arc::new(RpcStore {});
+    let store = Arc::new(DefaultStore {
+        store: store_client::ApiClient::connect(config.store_endpoint.to_string()).await?,
+    });
     let block_builder = DefaultBlockBuilder::new(store.clone());
     let batch_builder_options = DefaultBatchBuilderOptions {
         block_frequency: Duration::from_secs(10),
         max_batches_per_block: 4,
     };
-    let batch_builder = DefaultBatchBuilder::new(Arc::new(block_builder), batch_builder_options);
+    let batch_builder =
+        Arc::new(DefaultBatchBuilder::new(Arc::new(block_builder), batch_builder_options));
     let state_view = DefaulStateView::new(store.clone());
```

Comment:
typo: should be `DefaultStateView` rather than `DefaulStateView`.

Reply:
done